homeassistant/components/conversation/default_agent.py

```diff
@@ -422,16 +467,30 @@ def _recognize(
             language=language,
         ):
             if "name" in result.entities:
-                return result
+                name_result = result
 
-            # Keep looking in case an entity has the same name
-            maybe_result = result
-
-        if maybe_result is not None:
+            if (best_text_chunks_matched is None) or (
```

Comment:
What's this change doing? Can we extract this into a small PR?

Reply:
I'll pull it out into a small PR. It's for a corner case with wildcards I found while testing.